src/opm/output/eclipse/Summary.cpp

```diff
@@ -2642,6 +2673,145 @@ namespace Evaluator {
         }
     };
 
+    class InterRegionValue : public Base
+    {
+    public:
+        explicit InterRegionValue(Opm::EclIO::SummaryNode node,
+                                  const Opm::UnitSystem::measure m)
+            : node_   (std::move(node))
+            , m_      (m)
+            , regname_(node_.fip_region.has_value()
+                       ? node_.fip_region.value()
+                       : std::string{ "FIPNUM" })
+        {
+            this->analyzeKeyword();
+        }
+
+        void update(const std::size_t    /* sim_step */,
+                    const double            stepSize,
+                    const InputData&        input,
+                    const SimulatorResults& simRes,
+                    Opm::SummaryState&      st) const override
+        {
+            if (this->component_ == Component::NumComponents) {
+                return;
+            }
+
+            auto flows = simRes.ireg.find(this->regname_);
+            if (flows == simRes.ireg.end()) {
+                return;
+            }
+
+            auto flow = flows->second.getInterRegFlows(this->r1_, this->r2_);
+            if (! flow.has_value()) {
+                return;
+            }
+
+            const auto& usys = input.es.getUnits();
+            const auto  val  = this->getValue(flow->first, flow->second, stepSize);
+
+            updateValue(this->node_, usys.from_si(this->m_, val), st);
+        }
+
+    private:
+        using RateWindow = Opm::data::InterRegFlowMap::ReadOnlyWindow;
+        using Component  = RateWindow::Component;
+        using Direction  = RateWindow::Direction;
+
+        Opm::EclIO::SummaryNode node_;
+        Opm::UnitSystem::measure m_;
+        std::string regname_{};
+
+        Component component_{ Component::NumComponents };
+        Direction direction_{ Direction::Positive };
+        bool useDirection_{ false };
+        bool isCumulative_{ false };
+        int r1_{ -1 };
+        int r2_{ -1 };
+
+        void analyzeKeyword()
+        {
+            static const auto pattern = std::regex {
+                R"~~(R([OGW])F([RT])([-+GL_]?)([A-Z0-9_]{3})?)~~"
+            };
```

Comment:
This PR should have a test that makes sure that all keywords are correctly determined.

This regex gave me some headaches (that is probably me as I always struggle with regex). My observation is:
- It cannot be used to check whether a keyword is a inter-region flow (as there is quite some additional code to make that sure.
- Matching might create 4 groups, but the rest of the code only ever uses 3 of them.
- It seems to match `ROFR__AA`,  `ROFR_AA`,  `ROFR_A_A`, `ROFR_AA_` , `ROFRA_A`, `ROFRAA_` (is the underline really allowed to be everywhere in the last 3 characters?

That makes me a bit uneasy. A test might have helped to find a better one.
Judging from interregion:units and this regex it seems like `{R"~~(R([OGW])F([RT])([-+GL]|(?:_[A-Z0-9]{3}))?)~~"` might do the job. BUt I might be to naive here.

Reply:
If we're going to be really strict we need to be able to distinguish the following patterns

* `R[OGW]F[TR]`
* `R[OGW]F[TR][-+]`
* `R[OG]F[TR][GL]`
* `R[OGW]F[TR]_[A-Z0-9]{3}`
* `R[OGW]F[TR][-+][A-Z0-9]{3}`
* `R[OG]F[TR][GL][A-Z0-9]{3}`

and I'd really prefer not to have to list out all those alternatives.  It's also not **completely** clear to me if the underscore (`_`) can be part of the region set name.  These names are entered using the `FIP` keyword which doesn't really place any explicit restrictions on the characters used to name the region sets.  The only real restriction is that the first three characters after the string `FIP` must be unique among all `FIP*` keywords.

When there's no region name the inter-region flow keyword is four or five characters in length.  When there is a region name, the inter-region flow keyword is always exactly eight characters long with the final three (index 5-7 counting from zero) being the region set name and the fifth character (index 4 counting from zero) is either the underscore padding character (`_`), one of the direction indicators (`+` or `-`) or, in the case of oil/gas flow, one of the phase discriminators `L` or `G`.

Reply:
Maybe we should document this in the source?

Is there an intention to add a test?

Reply:
> Is there an intention to add a test?

I've pushed a basic update to `test_Summary` that exercises this code.  Note that our input reader and vector unit matcher will already have excluded summary keywords that don't meet the requirements for inter-region flows.  In other words, this section "only" needs to decompose the canonical keyword string to identify the relevant component and directionality, and whether or not we're calculating a cumulative volume or an instantaneous flow rate.

As part of creating the test I did find a bug in the original version.  We did not properly account for vaporised oil or dissolved gas when computing the "free oil" and "free gas" flows.  I've fixed that problem in the latest update to this PR.

Reply:
:sunglasses: